### PR TITLE
Add intersection of half-planes

### PR DESCRIPTION
**Pull request by Phil9l**
Add intersection of the half-planes method to geometry. Can be used to find the intersection of the convex polygons.

### CODE REVIEW

File: code/computational_geometry/halfplane_intersection/halfplane_intersection.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    for (int i = 0; i < 2; i++) {
+        Point p1, p2, p3;
+        p1.scan();
+        p2.scan();
+        p3.scan();
+        hpi.intersect(HalfPlane(Line(p1, p2), p3));
+        hpi.intersect(HalfPlane(Line(p1, p3), p2));
+        hpi.intersect(HalfPlane(Line(p2, p3), p1));
+    }
+    printf("%lf\n", hpi.perimeter());
```
Comment (arnavb) [inline code/computational_geometry/halfplane_intersection/halfplane_intersection.cpp]: Use cout

File: code/computational_geometry/halfplane_intersection/halfplane_intersection.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    }
+    void scan() {
+        scanf("%lf%lf", &x, &y);
+    }
+    void print() {
+        printf("%lf %lf\n", x, y);
+    }
+};
+
+struct Line {
```
Comment (arnavb) [inline code/computational_geometry/halfplane_intersection/halfplane_intersection.cpp]: class keyword makes more sense here for this class and the one below.

**Comment by Phil9l** [comment]
@arnavb, fixed.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**